src/DssSpell.t.sol

```diff
@@ -1600,27 +1856,148 @@ contract DssSpellTest is DSTest, DSMath {
         checkAuth(true);
     }
 
-    function test_core_unit_budgets() public {
-        address risk = 0xd98ef20520048a35EdA9A202137847A62120d2d9;
-        address multisig = 0x73f09254a81e1F835Ee442d1b3262c1f1d7A13ff;
+    function testSpellIsCast_RWA002_INTEGRATION_BUMP() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
 
-        uint256 prevSin = vat.sin(address(vow));
-        uint256 prevDai = dai.balanceOf(multisig);
-        uint256 prevRisk = dai.balanceOf(risk);
+        bumpSpell = new BumpSpell();
+        vote(address(bumpSpell));
 
-        assertEq(vat.can(address(pauseProxy), address(daiJoin)), 1);
+        bumpSpell.schedule();
 
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        (, address pip, ,) = oracle.ilks(ilk);
+
+        assertEq(DSValueAbstract(pip).read(), bytes32(5_634_804 * WAD));
+        bumpSpell.cast();
+        assertEq(DSValueAbstract(pip).read(), bytes32(5_634_805 * WAD));
+    }
+
+    function testSpellIsCast_RWA002_INTEGRATION_TELL() public {
         vote(address(spell));
-        spell.schedule();
-        castPreviousSpell();
-        hevm.warp(spell.nextCastTime());
-        spell.cast();
+        scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        assertEq(vat.can(address(pauseProxy), address(daiJoin)), 1);
+        tellSpell = new TellSpell();
+        vote(address(tellSpell));
+
+        tellSpell.schedule();
 
-        assertEq(vat.sin(address(vow)) - prevSin, 220_500 * RAD);
-        assertEq(dai.balanceOf(multisig) - prevDai, 120_000 * WAD);
-        assertEq(dai.balanceOf(risk) - prevRisk, 100_500 * WAD);
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        (, , uint48 tau, uint48 tocPre) = oracle.ilks(ilk);
+        assertTrue(tocPre == 0);
+        assertTrue(oracle.good(ilk));
+        tellSpell.cast();
+        (, , , uint48 tocPost) = oracle.ilks(ilk);
+        assertTrue(tocPost > 0);
```

Comment:
This could be more precisely set as `assertEq(tocPost, uint48(block.timestamp));`